deps/openvr/src/ivrsystem.cpp

```diff
@@ -841,12 +906,42 @@ NAN_METHOD(IVRSystem::TriggerHapticPulse)
     Nan::ThrowTypeError("Argument[2] must be a number.");
     return;
   }
+  if (!info[3]->IsNumber())
+  {
+    Nan::ThrowTypeError("Argument[3] must be a number.");
+    return;
+  }
   
-  vr::TrackedDeviceIndex_t unControllerDeviceIndex = info[0]->Uint32Value();
-  uint32_t unAxisId = info[1]->Uint32Value();
-  unsigned short usDurationMicroSec  = info[2]->Uint32Value();
+  int iIndex = info[0]->Int32Value();
+  uint32_t unAxisId = info[1]->Int32Value();
+  double dValue  = info[2]->NumberValue();
+  unsigned short usDurationMicroSec = info[3]->Uint32Value();
+  if (unAxisId >= 2) {
+    Nan::ThrowTypeError("Argument[1] must be 0 or 1.");
+    return;
+  }
 
-  obj->self_->TriggerHapticPulse(unControllerDeviceIndex, unAxisId, usDurationMicroSec);
+  if (iIndex < 0) {
```

Comment:
This seems very much a hack. Since this code is effectively bug openvr, can we move it to an entirely new dir?

Deciding this branch is probably best done higher in js, not in “openvr” (which it is apparently not).

Reply:
High praise. 

I disagree. I think that openvr is designed to support exactly this case, and their API is thoughtful and general. Exposing extra controllers for free is something everyone would want — why force devs to do extra work to support controllers, joysticks, steering wheels, and foot pedals? The same API can expose them all automatically.

Reply:
I agree the API is good and can be used. I just disagree that directx is a part of binding to openvr dlls, so the files could use a restructuring.